transformers/storage/cat/repository.go

```diff
@@ -51,23 +51,38 @@ func (repository *StorageRepository) SetDB(db *postgres.DB) {
 }
 
 func (repository *StorageRepository) insertLive(diffID, headerID int64, live string) error {
-	_, err := repository.db.Exec(insertCatLiveQuery, diffID, headerID, live)
+	addressID, addressErr := shared.GetOrCreateAddress(repository.ContractAddress, repository.db)
```

Comment:
Given the number of times that these inserts can happen, I think it might be cool to have a singleton-ish helper function that memoizes the addressID the first time it's called - so that we don't need to go to the DB for it every time

Reply:
fwiw we already do what I'm suggesting here: https://github.com/makerdao/vdb-mcd-transformers/blob/staging/transformers/storage/median/repository.go#L292

Reply:
That makes sense to me - originally I had this passing around the ID but I didn't care for that.